geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/AcceptorImpl.java

```diff
@@ -1030,22 +1094,34 @@ private void checkForStuckKeys() {
         // just ignore this channel and try the next one
         finishCon(sc);
       } catch (IOException | NullPointerException ex) {
-        if (isRunning() && selector.isOpen() && tmpSel.isOpen()) {
+        if (isRunning() && selector[index].isOpen() && tmpSel[index].isOpen()) {
           logger.warn("Unexpected Exception:", ex);
         }
       } // fix bug 39644
 
     }
   }
 
-  private int registeredKeys;
-
   private void runSelectorLoop() {
     // int zeroEventsCount = 0;
+    String regex = "#([0-9]*)";
+    Pattern p = Pattern.compile(regex);
+    Matcher m = p.matcher(Thread.currentThread().getName());
+    int myIndex = 0;
+    if (m.find()) {
+      try {
+        String num = m.group(0).substring(1).trim();
+        myIndex = Integer.parseInt(num);
+      } catch (Exception e) {
+        logger.info("Not able to extract selector thread ID");
+        throw new RuntimeException("Not able to extract selector thread ID");
+      }
+    }
+    registeredKeys[myIndex] = 0;
```

Comment:
This code is unnecessarily complex. Why not pass the index to this runSelectorLoop() method? That way you get rid of the regex, of parsing the thread name and of the possible strange Exception you could get.

Reply:
I agree about the code complexity here, but the fact that it has already been tested for XXL is a valid fact also in this case.

If we go for some major code changes or refactoring here, I think that XXL tests performed for this feature would need to be repeated.

Reply:
I don't agree with the philosophy followed in this pull request. Anything we add to the repo must meet the functional requirements as well as the quality requirements.
We are allowing to include code with not the best quality because it has already been tested.

Reply:
As agreed and mentioned in previous comment:

all code refactoring - where it is currently to complex without real necessity - and adding functionality like e.g. to have a general solution for all stats - we will do - but for the next release